tests/unit/test_signature.cpp

```diff
@@ -0,0 +1,427 @@
+ * - Sign a plain text message using a signature context
+ * - Sign a message digest using a signature context
+ * - Sign a plain text message using a private key
+ * - Sign a message digest using a private key
+ */
+TEST_F(SignatureTest, testSignaturePKCS1)
+{
+    std::vector<uint8_t> signature;
+    SignatureUtils::Context<AsymmetricPrivateKey> signatureContext(
+          mococrw::AsymmetricKeypair::readPrivateKeyFromPEM(_validPrivateKey, ""));
+
+    /* Tests the interface that requires an plain text message and a context to be set and passed */
+    signatureContext.setPadding(openssl::RSAPaddingMode::PKCS1);
+
+    EXPECT_NO_THROW(signature = SignatureUtils::create(signatureContext, _testMessage));
+    signatureContext.setHashingFunction(openssl::DigestTypes::SHA256);
+    EXPECT_EQ(_validPKCS1SignatureSHA256, signature);
+    signatureContext.setHashingFunction(openssl::DigestTypes::SHA512);
+    EXPECT_NO_THROW(signature = SignatureUtils::create(signatureContext, _testMessage));
+    EXPECT_EQ(_validPKCS1SignatureSHA512, signature);
+
+    /* Tests the interface that requires a message digest and a context to be set and passed */
+    /* The padding in the context is by default SHA256, so there is no need to change the context */
+    signatureContext.setHashingFunction(openssl::DigestTypes::SHA256);
+    EXPECT_NO_THROW(signature = SignatureUtils::create(signatureContext, _testMessageDigestSHA256));
+    EXPECT_EQ(_validPKCS1SignatureSHA256, signature);
+    signatureContext.setHashingFunction(openssl::DigestTypes::SHA512);
+    EXPECT_NO_THROW(signature = SignatureUtils::create(signatureContext, _testMessageDigestSHA512));
+    EXPECT_EQ(_validPKCS1SignatureSHA512, signature);
+
+    /* Tests the interface that takes a private key and a message digest */
+    signatureContext.setHashingFunction(openssl::DigestTypes::SHA256);
+    EXPECT_NO_THROW(signature = SignatureUtils::create(
+                          mococrw::AsymmetricKeypair::readPrivateKeyFromPEM(_validPrivateKey, ""),
+                          _testMessageDigestSHA256));
+    EXPECT_EQ(_validPKCS1SignatureSHA256, signature);
+    signatureContext.setHashingFunction(openssl::DigestTypes::SHA512);
+
+    /* Tests the interface that takes a private key and creates a signature using default values */
+    signatureContext.setHashingFunction(openssl::DigestTypes::SHA256);
+    EXPECT_NO_THROW(signature = SignatureUtils::create(
+                          mococrw::AsymmetricKeypair::readPrivateKeyFromPEM(_validPrivateKey, ""),
+                          _testMessageDigestSHA256));
+    EXPECT_EQ(_validPKCS1SignatureSHA256, signature);
+}
+
+/**
+ * @brief Successful signature creation and verification
+ * Cases:
+ * - Sign and verify using PKCS1
+ * - Sign and verify using PCSS
+ * - Sign and verify using SHA512 hashing
+ * - Sign and verify using SHA512 masking
+ * - Sign and verify using non-default salt length
+ */
+TEST_F(SignatureTest, testSignatureSuccessfulVerification)
```

Comment:
It might be a good idea to split this big test into several test-cases. In case of failures, the bug would be better isolated and we would not reuse the contexts (which might have side-effects).